component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -935,7 +935,14 @@ public boolean sendRESTCall(AuthenticationContext context, String smsUrl, String
         String encodedMobileNo = encoder.encode(mobile);
         smsUrl = smsUrl.replaceAll("\\$ctx.num", encodedMobileNo).replaceAll("\\$ctx.msg",
                 smsMessage.replaceAll("\\s", "+") + otpToken);
-        URL smsProviderUrl = new URL(smsUrl);
+        URL smsProviderUrl = null;
+        try {
+            smsProviderUrl = new URL(smsUrl);
+        } catch (MalformedURLException e) {
+            log.error("Invalid SMS URL provided : " + e);
+            context.setProperty(SMSOTPConstants.ERROR_CODE, "Invalid SMS URL provided");
```

Comment:
Can we change this to "The SMS URL does not conform to URL specification"

Reply:
+1 Let's change this